online-shop/src/app/login/login.component.ts

```diff
@@ -43,8 +42,8 @@ export class LoginComponent implements OnInit {
         username: this.myForm.value.username,
         password: this.myForm.value.password
       }
-  
-      this.loginSubscription = this.loginService.login(user.username, user.password).subscribe(() =>  {const redirectUrl = this.loginService.redirectUrl;
+      this.store.dispatch(loginUser({username: user.username, password: user.password}));
+     this.loginSubscription = this.loginService.login(user.username, user.password).subscribe(() =>  {const redirectUrl = this.loginService.redirectUrl;
         this.router.navigateByUrl(RoutesEnum.ProductList);
       });
```

Comment:
Instead of this, you can create a login action, reducer and selector to set the redirect url. And then by having a subscribtion to the `loginUserSuccess`, the customer can be redirected to the `RoutesEnum.ProductList`, because there should already be in place the effect of `this.store.dispatch(loginUser({username: user.username, password: user.password}));`. Therefore not calling the login method 2 times